### ISSUE TITLE
W1b: `OperationBase`'s symmetric magnitude multiplies by a bool, so half of every `RandAugment` batch gets 0

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4440]
`kornia.augmentation.auto.rand_augment.ops` builds `rotate`, `shear_x`, `shear_y`, `translate_x` and `translate_y`
with `symmetric_megnitude=True`, and each of those factories rejects an asymmetric range first
(`if min_mag != -max_mag: raise ValueError(...)`, `auto/rand_augment/ops.py:76-83` for `rotate`), so a `RandAugment`
policy entry written `("rotate", -30.0, 30.0)` promises rotation in both directions.

It never rotates in the negative direction. `OperationBase._init_magnitude_fn`'s `_random_flip`
(`auto/operations/base.py:82-86`) draws `flip = torch.rand((x.shape[0],), device=x.device) > 0.5` — a **bool**
tensor — and returns `fn(x) * flip`. Multiplying by a bool keeps the magnitude where `flip` is true and sets it to
**zero** where it is false, instead of negating it. So those five operations draw from `{0, +mag}` rather than
`{-mag, +mag}`: the negative half of the advertised range is unreachable, and roughly half of every batch receives
magnitude zero, which for a geometric operation is the identity. At `B = 256` the zero fraction is close to one
half and reads as a weaker augmentation; at smaller batch sizes it is a lottery, and a small batch can come out
entirely unaugmented for that operation.

## Repro

```python
import torch
import kornia.augmentation.auto as AA
from kornia.augmentation.auto.operations.base import OperationBase
CASES = [("rotate", -30.0, 30.0, "degrees"), ("shear_x", -0.3, 0.3, "shear_x"), ("shear_y", -0.3, 0.3, "shear_y"),
         ("translate_x", -0.5, 0.5, "translate_x"), ("translate_y", -0.5, 0.5, "translate_y")]

def drawn(name, lo, hi, key, m, B, seed):
    torch.manual_seed(seed)
    aug = AA.RandAugment(n=1, m=m, policy=[[(name, lo, hi)]])
    aug(torch.rand(B, 3, 16, 16))
    return aug._params[0].data[0].data[key].flatten()

for B, m, seed in ((256, 29, 7), (64, 1, 7)):
    print(f"--- B={B}, m={m}, seed={seed} ---")
    for name, lo, hi, key in CASES:
        v = drawn(name, lo, hi, key, m, B, seed)
        print(f"{name:12s} unique={[round(u, 4) for u in v.unique().tolist()]}"
              f"  zero_frac={(v == 0).float().mean():.3f}  neg_frac={(v < 0).float().mean():.3f}")

print("--- a whole small batch can be the identity: m=15, B=2 ---")
for seed in (7, 0, 3):
    v = drawn("rotate", -30.0, 30.0, "degrees", 15, 2, seed)
    print(f"seed={seed}  degrees={[round(u, 4) for u in v.tolist()]}  zero_frac={(v == 0).float().mean():.3f}")

print("--- which of the three policy classes reaches _random_flip ---")
torch.manual_seed(7); x = torch.rand(64, 3, 16, 16)
for label, aug in (("RandAugment", AA.RandAugment(n=1, m=29, policy=[[("rotate", -30.0, 30.0)]])),
                   ("TrivialAugment", AA.TrivialAugment(policy=[[("rotate", -30.0, 30.0)]])),
                   ("AutoAugment bin=0", AA.AutoAugment(policy=[[("rotate", 1.0, 0)]])),
                   ("AutoAugment bin=9", AA.AutoAugment(policy=[[("rotate", 1.0, 9)]]))):
    aug(x)
    d = aug._params[0].data[0].data["degrees"]
    flags = [m.symmetric_megnitude for m in aug.modules() if isinstance(m, OperationBase)]
    print(f"{label:18s} symmetric_megnitude={flags}  degrees min={d.min():.4f} max={d.max():.4f}"
          f"  zero_frac={(d == 0).float().mean():.3f}")
```

## Observed

```
--- B=256, m=29, seed=7 ---
rotate       unique=[0.0, 29.0]  zero_frac=0.484  neg_frac=0.000
shear_x      unique=[0.0, 52.2]  zero_frac=0.465  neg_frac=0.000
shear_y      unique=[0.0, 52.2]  zero_frac=0.465  neg_frac=0.000
translate_x  unique=[0.0, 0.4833]  zero_frac=0.484  neg_frac=0.000
translate_y  unique=[0.0, 0.4833]  zero_frac=0.484  neg_frac=0.000
--- B=64, m=1, seed=7 ---
rotate       unique=[0.0, 1.0]  zero_frac=0.516  neg_frac=0.000
shear_x      unique=[0.0, 1.8]  zero_frac=0.500  neg_frac=0.000
shear_y      unique=[0.0, 1.8]  zero_frac=0.500  neg_frac=0.000
translate_x  unique=[0.0, 0.0167]  zero_frac=0.516  neg_frac=0.000
translate_y  unique=[0.0, 0.0167]  zero_frac=0.516  neg_frac=0.000
--- a whole small batch can be the identity: m=15, B=2 ---
seed=7  degrees=[0.0, 0.0]  zero_frac=1.000
seed=0  degrees=[15.0, 0.0]  zero_frac=0.500
seed=3  degrees=[0.0, 0.0]  zero_frac=1.000
--- which of the three policy classes reaches _random_flip ---
RandAugment        symmetric_megnitude=[True]  degrees min=0.0000 max=29.0000  zero_frac=0.516
TrivialAugment     symmetric_megnitude=[True]  degrees min=0.1085 max=29.6658  zero_frac=0.000
AutoAugment bin=0  symmetric_megnitude=[False]  degrees min=-29.9259 max=-24.0319  zero_frac=0.000
AutoAugment bin=9  symmetric_megnitude=[False]  degrees min=24.0179 max=29.8717  zero_frac=0.000
```

## Expected

`unique = [-mag, 0.0, mag]` at most, with `neg_frac` near `0.5` and `zero_frac` near zero — the symmetric range the
factories validate for. The shape of the fix is to make `flip` a sign rather than a mask, for instance
`torch.where(torch.rand(...) > 0.5, 1.0, -1.0)`, so `fn(x) * flip` negates.

The blast radius is `RandAugment` alone, which the last block of the repro settles by execution over the three
policy classes. Only `RandAugment` both builds its operations with `symmetric_megnitude=True` and draws them
through `OperationBase.forward_parameters`, the path that applies `_random_flip`. `TrivialAugment` reuses the same
`rand_augment.ops` factories so the flag is `True` there too, but draws through `PolicySequential` and never
reaches `_random_flip`; its one-sided magnitudes have a different cause, filed separately in this batch.
`AutoAugment` builds every geometric operation with `symmetric_megnitude=False`
(`auto/autoaugment/ops.py:48, 59, 70, 81, 92`) and reaches negative magnitudes through signed bins: bin `0` of
`rotate` draws inside `[-30, -24]`. Fixing `_random_flip` leaves both of them byte-identical, and changes every
current `RandAugment` result — strictly more augmentation, in the opposite direction half the time — so it is a
behavior change. `Rotate`'s own guard (`symmetric_megnitude and magnitude_range[0] < 0` raises) assumes the range
passed down is the non-negative half and stays correct under the fix.

## Classification

**Broken output** — five operations that validate for a symmetric range produce a one-sided one, and about half of
every batch is an identity for them. Idiom: 3 (a guard whose predicate does not match the documented contract —
the factories validate `-min_mag == max_mag` for a symmetry the magnitude function does not implement).

## Disposition

**MERGE IN WINDOW — behavior change.** Canonical: `_random_flip` in `kornia/augmentation/auto/operations/base.py`,
where all five `rand_augment.ops` operations get their sign. Every `RandAugment` result changes, so the decision
belongs in the coordinated Repair & Deprecation window rather than in a drive-by PR. Batch 6 documents the current
one-sided draw on `OperationBase` and pins it, and does not change it.

## Environment

kornia 0.9.0rc1 @ `d6a2c138` (main), torch 2.14.0, python 3.11, cpu. Found during the augmentation conventions
audit (batch 6, tracking issue #4407); the batch documents the behavior above as it is and pins it, and does not
change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
